### PR TITLE
Handle Destiny Uniqueness Violation

### PR DESCRIPTION
**Pull request by kyleshay**
Trying to close out #79
 
screen shot 2016-05-04 at 12 55 58 pm

screen shot 2016-05-04 at 12 56 15 pm

maybe .toLower on `Emote`? 

Have a better name for the warning title? 

Should it still be considered an error popup versus warning?

### CODE REVIEW

File: app/scripts/services/dimBungieService.factory.js [hunk 1]
```diff
@@ -406,11 +406,33 @@
...
+          item.type,
+          "to",
+          store.isVault ?
+            'the vault' :
+            'your ' + store.powerLevel + ' ' + store.race + ' ' + store.name,
+          "but that destination already has that",
+          item.type,
+          "and is only allowed one."
+        ].join(' '));
+        return $q.reject(new Error('move-canceled'));
```
Comment (kyleshay) [inline app/scripts/services/dimBungieService.factory.js]: @bhollis is this solution cool? or should i build it into the `handleErrors` function?

**Comment by kyleshay** [comment]
to lowered the `item.type`

 > You tried to move the 'Point' emote to your 330 exo warlock but that destination already has that item and is only allowed one.


screen shot 2016-05-04 at 5 10 39 pm

- [ ] 👍  
- [ ] 👎  
- [ ] 😭

File: app/scripts/services/dimBungieService.factory.js (hunk 1 above)
Comment (kyleshay) [inline app/scripts/services/dimBungieService.factory.js]: i need access to item, store. but i could make `handleErrors` accept `response` and `data`. data could just be an object w/ whatever on it. handleErrors would have to know how to deal with it. not sure which way i like more.

File: app/scripts/services/dimBungieService.factory.js [hunk 2]
```diff
@@ -406,11 +406,31 @@
...
+    //Handle "DestinyUniquenessViolation" (1648)
+    function handleUniquenessViolation(response, item, store) {
+      if (response && response.data && response.data.ErrorCode === 1648) {
+        toaster.pop('warning', 'Item Uniqueness', [
+          "You tried to move the '" + item.name + "'",
+          item.type.toLowerCase(),
+          "to",
+          store.isVault ?
+            'the vault' :
+            'your ' + store.powerLevel + ' ' + store.race + ' ' + store.name,
```
Comment (bhollis) [inline app/scripts/services/dimBungieService.factory.js]: I've been doing race + name, but not power level. We should standardize, whatever we do.

**Comment by bhollis** [comment]
This looks great. Up to you if you want to remove the power level from the text, but merge away whenever.

**Comment by SunburnedGoose** [comment]
Power Level is helpful until everyone only has 335 gear :)

**Comment by bhollis** [comment]
Ehh, don't tell me you created two characters of the same class and race!?

**Comment by SunburnedGoose** [comment]
Maybeh...

**Comment by kyleshay** [comment]
> We should standardize, whatever we do. 

agreed. could maybe just make a property/function on the character object and just call that in places instead of appending it manually in each.

**Pull request merged**